docs/spec/spec.md

```diff
@@ -0,0 +1,310 @@
+| referenceCount | Integer | Number of Owners for this EventType |             |
+
+##### Conditions
+
+- **Pinned**. True when the EventType cannot be deleted.
+
+#### Events
+
+- **Owned.** When EventType has a new Provisioner Owner.
+- **Released.** When a Provisioner removes Ownership from the EventType.
+
+### Life Cycle
+
+| Action | Reactions | Limitations                                |
+| ------ | --------- | ------------------------------------------ |
+| Create |           |                                            |
+| Update |           |                                            |
+| Delete |           | Blocked until all provisioners release it. |
+
+---
+
+## Shared Object Schema
+
+### ProvisionerReference
+
+| Field                | Type            | Description | Limitations                   |
+| -------------------- | --------------- | ----------- | ----------------------------- |
+| ref<sup>1</sup>      | ObjectReference |             |                               |
+| selector<sup>1</sup> | LabelSelector   |             | Must match only one resource. |
+
+1: One of (name, selector), Required.
+
+### EndpointSpec
+
+| Field                 | Type            | Description | Limitations                |
+| --------------------- | --------------- | ----------- | -------------------------- |
+| targetRef<sup>1</sup> | ObjectReference |             | Must adhere to Targetable. |
+| dnsName<sup>1</sup>   | String          |             |                            |
+
+1: One of (targetRef, dnsName), Required.
+
+### ProvisionedObjectStatus
+
+| Field    | Type   | Description                                                  | Limitations |
+| -------- | ------ | ------------------------------------------------------------ | ----------- |
+| name\*   | String | Name of Object                                               |             |
+| type\*   | String | Fully Qualified Object type.                                 |             |
+| status\* | String | Current relationship between Provisioner and Object.         |             |
+| reason   | String | Detailed description describing current relationship status. |             |
+
+\*: Required
+
+### Subscribable
+
+| Field       | Type            | Description                      | Limitations |
+| ----------- | --------------- | -------------------------------- | ----------- |
+| channelable | ObjectReference | The channel used to emit events. |             |
+
+### Channelable
+
+| Field       | Type                    | Description                                                           | Limitations                            |
+| ----------- | ----------------------- | --------------------------------------------------------------------- | -------------------------------------- |
+| subscribers | ChannelSubscriberSpec[] | Information about subscriptions used to implement message forwarding. | Filled out by Subscription Controller. |
```

Comment:
Do we mean to nest this one extra level deep? i.e. `spec.channelable.subscribers` vs `spec.subscribers`?

Reply:
We did. I checked with @vaikas-google

Reply:
Do we expect a Knative Service to implement both `.status.targetable.domainInternal` and `.status.sinkable.domainInternal`?

Reply:
I think we're keeping this resolved `ChannelSubscriberSpec[]` as the definiton of Subscribable, but only implementing it on Channel for now. Separating out the resolution of endpoints (which we may want to iterate on separately from all Channel implementations) makes sense to me.

Reply:
added `channel.spec.subscribers`. we can revisit later and move to fanout or something else